Automattic/wp-calypso
# Purchases: Update me/purchases module to ECMAScript 6

This pull request updates the `client/me/purchases` section to use as many [ES6](https://github.com/Automattic/calypso-pre-oss/blob/master/docs/coding-guidelines/javascript.md#es6) features as possible. It also fixes and improves a few minor things in the process - such as missing props validation.
 
![screenshot](https://cloud.githubusercontent.com/assets/594356/14077217/be9c28e8-f4e9-11e5-978a-66cc6d8554cb.png)

#### Additional notes
 
You should probably consider browsing commits [one by one](https://github.com/Automattic/wp-calypso/pull/4346/commits) to review this pull request since it might be easier to understand the changes rather than in the [unified diff view](https://github.com/Automattic/wp-calypso/pull/4346/files).

#### Testing instructions
 
1. Run `git checkout update/me-purchases-to-es6` and start your server
2. Purchase different types of products, using different methods of payment
3. Check that the [`Purchases` section](http://calypso.localhost:3000/purchases) behave as before

#### Reviews
